### PR TITLE
test(connector): kafka source test with schema registry

### PR DESCRIPTION
**Pull request by idx0-dev**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Our E2E tests leak support for avro for not containing a schema registry in environment. This PR has replace the original kafka with redpanda, which has builtin support for schema registry.

## Checklist For Contributors

~~- [ ] I have written necessary rustdoc comments~~
- [x] I have added necessary unit tests and integration tests
~~- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).~~
~~- [ ] I have demonstrated that backward compatibility is not broken by breaking changes and created issues to track deprecated features to be removed in the future. (Please refer to the issue)~~
~~- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)~~

## Checklist For Reviewers

~~- [ ] I have requested macro/micro-benchmarks as this PR can affect performance substantially, and the results are shown.~~

## Documentation

- [x] My PR **DOES NOT** contain user-facing changes.

<details><summary>Click here for Documentation</summary>

### Types of user-facing changes

Please keep the types that apply to your changes, and remove the others.

- Installation and deployment
- Connector (sources & sinks)
- SQL commands, functions, and operators
- RisingWave cluster configuration changes
- Other (please specify in the release note below)

### Release note

</details>

### CODE REVIEW

File: scripts/source/prepare_ci_kafka.sh [hunk 1]
```diff
@@ -63,6 +63,8 @@ for filename in $kafka_data_files; do
     # binary data, one message a file, filename/topic ends with "bin"
     if [[ "$topic" = *bin ]]; then
         ${KCAT_BIN} -P -b 127.0.0.1:29092 -t "$topic" "$filename"
+    elif [[ "$topic" = *avro_json ]]; then 
+        python3 avro_producer.py "127.0.0.1:29092" "http://127.0.0.1:8081" "$filename"
```
Comment (tabVersion) [inline scripts/source/prepare_ci_kafka.sh:67]: no change to ci/docker-compose.yml?

File: e2e_test/source/basic/nosim_kafka.slt [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Start with nosim to avoid running in deterministic test
+
+
+# key schema should be a subset of value schema
+statement error
+CREATE TABLE upsert_avro_json ()
+WITH (
+	connector = 'kafka',
+  properties.bootstrap.server = 'message_queue:29092',
+	topic = 'upsert_avro_json')
```
Comment (tabVersion) [inline e2e_test/source/basic/nosim_kafka.slt:10]: what upsert_avro_json means?

File: e2e_test/source/basic/nosim_kafka.slt [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+# statement ok
+# CREATE TABLE upsert_avro_json (
+#    id TEXT  PRIMARY KEY
+# )
+# WITH (
+#	connector = 'kafka',
+#  properties.bootstrap.server = 'message_queue:29092',
+#	topic = 'upsert_avro_json')
+# ROW FORMAT UPSERT_AVRO 
+#row schema location confluent schema registry 'http://message_queue:8081'
```
Comment (tabVersion) [inline e2e_test/source/basic/nosim_kafka.slt:24]: may remove discarded tests

**Review by tabVersion — APPROVED** [review]

File: e2e_test/source/basic/nosim_kafka.slt [hunk 4]
```diff
@@ -0,0 +1,47 @@
+# Start with nosim to avoid running in deterministic test
```
Comment (xxchan) [inline e2e_test/source/basic/nosim_kafka.slt:1]: Does this really work?

**Pull request merged**